src/smc-webapp/r_misc.cjsx

```diff
@@ -1442,18 +1442,52 @@ exports.UpgradeAdjustor = rclass
         return state
 
     get_quota_info : ->
+        # This function is quite confusing and tricky.
+        # It combines the remaining upgrades of the user with the already applied ones by the same user.
+        # Then it limits the applyable upgrades by what's still possible to apply until the maximum is reached.
+        # My mental model:
+        #
+        #   0                                total          maximum
+        #   |<-------------------------------->|                |
+        #   |<----->|<------------------------>|<-------------->|
+        #   | admin |  all upgrades by users   | proj remainder |
+        #   | +     |<------------>|<--------->|<--------->|    |
+        #   | free  |  other users | this user | remaining |    |
+        #   |       |              |           | this user |    |
+        #   |       |              |<--------------------->|    |
+        #   |       |              |  limit for this user  | <= | max
+        #
+        #   admin/free: could be 0
+        #   all upgrades by users is total_project_quotas
```

Comment:
This definition and line 1479 `proj_remainder = misc.map_diff(proj_maximum, total_upgrades)` are conflicting. Based on the correct behavior, it seems `total_project_quotas` is actually `total` in this diagram above.

@haraldschilly Please confirm if this is true.

Reply:
I pushed a slight change, such that the variable names show up in the diagram. I hope I haven't confused myself again when looking at it :face_with_head_bandage: